Implemented code for gitHub ticket #642

sumitzarkar opened the pull request:
We have implemented code for  following points:
 1. Removing: ArcMapAddinCoordinateConversion project/dll
 2. Removing:CoordinateConversionLibrary project/dll
 3. Refactoring the Pro project to remove dependencies on former CoordinateConversionLibrary
   a. remove view inheritance
  b. remove no-longer-needed classes (super/abstract classes)
  c. remove mediator/dispatcher (ex: convert to RelayCommands)
  d. use Pro styling for Dark Mode support
5. As part of this if you notice any hard-coded UI strings that should be converted to Resources (for I18n) please convert these to resources

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint1/2

Implemented following points:
 1. Removing: ArcMapAddinCoordinateConversion project/dll
 2. Removing:CoordinateConversionLibrary project/dll
 3. Refactoring the Pro project to remove dependencies on former CoordinateConversionLibrary
    a. remove view inheritance

Note: This build is work in progress and some build errors can be observed

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint2/2

- Completed refactoring

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint 2/2 Bug fixes

- Fixed crash found after last commit

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint 2/2 - Pro Styles

Updated code to make Pro Style Changes.

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint 2/2 - Localization & Pro Styles for Buttons

Implemented code for gitHub ticket #642 Sprint 2/2 - Localization & Pro Styles for Buttons

csmoore committed:
Add resource dictionary

csmoore commented:
I had a small build issue - not sure why it only showed up for me, but fixed in https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/645 

I also now see that the Units Tests don't run because of the new Pro dependencies - I'll try to figure out how to get these tests running for acceptance (like maybe just put these tests -  `CoordinateConversion\CoordinateConversionLibrary.Tests` - in a quick addin just so we can run them)

sumitzarkar committed:
Merge pull request #645 from Esri/CT-Ticket642-Edits

Add resource dictionary

jmccausland commented on source/CoordinateConversion/ProAppCoordConversionModule/ViewModels/TabBaseViewModel.cs:
```diff
@@ -43,10 +44,14 @@ public TabBaseViewModel()
+
+            //Anupam
+            NewMapPointInternal = new RelayCommand(OnNewMapPointInternal);
+            ValidateMapPointInternal = new RelayCommand(OnValidateMapPointInternal);
```
@csmoore will cause validation to be run on each derived tab, that would be 2 in its current state.  Mediator is a generic version of CompositePresentationEvent https://pro.arcgis.com/en/pro-app/sdk/api-reference/#topic7802.html You can reproduce this visually by zooming out beyond the map extent and clicking outside of said extent and see the warning dialog twice.

Events avoids the need for view models to know about each other and don't break the pattern of MVVM like relay commands use here does.

jmccausland commented on source/CoordinateConversion/ProAppCoordConversionModule/ViewModels/TabBaseViewModel.cs:
```diff
@@ -25,9 +25,10 @@
+namespace ProAppCoordConversionModule.ViewModels
 {
-    public class TabBaseViewModel : BaseViewModel
+    public class TabBaseViewModel : NotificationObject
```
Instead of NotificationObject, I would use the Pro sdk provided ViewModelBase https://pro.arcgis.com/en/pro-app/sdk/api-reference/#topic10691.html

jmccausland commented on source/CoordinateConversion/ProAppCoordConversionModule/Config.daml:
```diff
@@ -1,11 +1,12 @@
     <Author>Esri</Author>
     <Company>Esri</Company>
-    <Date>6/26/2018</Date>
+    <Date>2/26/2020</Date>
```
since the DAML is doing this `<updateModule refID="esri_mapping">`  you probably should have core and mapping added as a dependencies like the following:

`  <dependencies>
    <dependency name="ADCore.daml" />
    <dependency name="ADMapping.daml" />
  </dependencies>
`

I know in other cases there can be a race condition and the update is performed before the other daml is loaded resulting in no update.

csmoore commented:
@sumitzarkar Sorry for my delay reviewing this PR - @OscarVilla1 tested successfully and I think the only issue I noticed was that there are 2 base classes that could be combined into one class/file:

1. TabBaseViewModel and ProTabBaseViewModel
2. OutputCoordinateViewModel and OutputCoordinateViewModel

csmoore commented:
Opened alternate PR #646 - so we could just switching the head branch we are merging  into (branch: **Pro2.6** vs. dev). Feel free to move any discussion to that PR and sorry for any confusion.

sumitzarkar committed:
Implemented code review comments for #642

Implemented following points:
 1. Instead of NotificationObject, use the Pro sdk provided ViewModelBase
 2. Add the dependencies in Config.daml

csmoore commented:
Closing unmerged to merge into different head branch in PR #646
